tooling/metamodel-generator/src/main/java/org/hibernate/processor/annotation/NonManagedMetamodel.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.processor.annotation;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.hibernate.processor.Context;
+
+import javax.lang.model.element.TypeElement;
+
+public class NonManagedMetamodel extends AnnotationMetaEntity {
+
+	public NonManagedMetamodel(TypeElement element, Context context, boolean jakartaDataStaticMetamodel, @Nullable AnnotationMeta parent) {
+		super( element, context, false, jakartaDataStaticMetamodel, parent );
+	}
+
+	public static NonManagedMetamodel create(
+			TypeElement element, Context context,
+			boolean jakartaDataStaticMetamodel,
+			@Nullable AnnotationMetaEntity parent) {
+		final NonManagedMetamodel metamodel =
+				new NonManagedMetamodel( element, context, jakartaDataStaticMetamodel, parent );
+		if ( parent != null ) {
+			metamodel.setParentElement( parent.getElement() );
+			parent.addInnerClass( metamodel );
+		}
+		return metamodel;
+	}
+
+	protected void init() {
```

Comment:
@cigaly Can you add the missing `@Override` please?

Reply:
Done! Local tests are passing, so I guess that I did not messed anything :-)